invenio/modules/jsonalchemy/manage.py

```diff
@@ -0,0 +1,101 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Invenio.
+# Copyright (C) 2015 CERN.
+#
+# Invenio is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# Invenio is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Invenio; if not, write to the Free Software Foundation, Inc.,
+# 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+
+"""Generate JSONSchema output from existing JSONAlchemy models."""
+
+from __future__ import print_function
+
+import json
+
+from flask import current_app
+
+from invenio.ext.script import Manager
+
+manager = Manager(usage=__doc__)
+
+
+@manager.command
+@manager.option('-n', '--namespace', dest='namespace', default='recordext',
+                help="Desired namespace")
+@manager.option('-m', '--model', dest='model', default='__default__',
+                help="Desired model. e.g. 'hep'")
```

Comment:
Mmh... I tried as you said (I never used `argparse` before), but somehow the magic doesn't seem to work. `output_file` is initialized with the name of the file instead of the actual opened file.

Do you know what can be wrong?

```python
Traceback (most recent call last):
  File "/home/skaplun/.virtualenvs/master/bin/inveniomanage", line 9, in <module>
    load_entry_point('invenio==2.2.0.dev20150608', 'console_scripts', 'inveniomanage')()
  File "/home/skaplun/.virtualenvs/master/src/invenio/invenio/base/manage.py", line 103, in main
    manager.run()
  File "/home/skaplun/.virtualenvs/master/local/lib/python2.7/site-packages/flask_script/__init__.py", line 412, in run
    result = self.handle(sys.argv[0], sys.argv[1:])
  File "/home/skaplun/.virtualenvs/master/local/lib/python2.7/site-packages/flask_script/__init__.py", line 383, in handle
    res = handle(*args, **config)
  File "/home/skaplun/.virtualenvs/master/src/invenio/invenio/ext/script/__init__.py", line 177, in __call__
    res = super(SignalingCommand, self).__call__(*args, **kwargs)
  File "/home/skaplun/.virtualenvs/master/local/lib/python2.7/site-packages/flask_script/commands.py", line 216, in __call__
    return self.run(*args, **kwargs)
  File "/home/skaplun/.virtualenvs/master/src/invenio/invenio/modules/jsonalchemy/manage.py", line 93, in jsonschema
    output_file.write(json.dumps(out, indent=4, sort_keys=True))
AttributeError: 'unicode' object has no attribute 'write'
```